pkg/controller/dgraph/models/query/constants_test.go

```diff
@@ -0,0 +1,657 @@
+			memoryCost: memoryCostPod as math(memoryPod * durationInHoursPod * pricePerMemoryPod)
+			storageCost: storageCostPod as math(storagePod * durationInHoursPod * 0.00013888888)`
+
+const testQueryForAggregatingChildMetricsWithAlias = `name
+			type
+			cpu: sum(val(cpuPod))
+			memory: sum(val(memoryPod))
+			storage: sum(val(storagePod))
+			cpuCost: sum(val(cpuCostPod))
+			memoryCost: sum(val(memoryCostPod))
+			storageCost: sum(val(storageCostPod))`
+
+const testQueryForPodParentMetrics = `query {
+		parent(func: has(isJob)) @filter(eq(name, "job-purser")) {
+			children: ~job @filter(has(isPod)) {
+				name
+			type
+			cpu: cpuPod as cpuRequest
+			memory: memoryPod as memoryRequest
+			storage: storagePod as storageRequest
+			stPod as startTime
+			stSecondsPod as math(since(stPod))
+			secondsSinceStartPod as math(cond(stSecondsPod > 1.45, 1.45, stSecondsPod))
+			etPod as endTime
+			isTerminatedPod as count(endTime)
+			secondsSinceEndPod as math(cond(isTerminatedPod == 0, 0.0, since(etPod)))
+			durationInHoursPod as math(cond(secondsSinceStartPod > secondsSinceEndPod, (secondsSinceStartPod - secondsSinceEndPod) / 3600, 0.0))
+			pricePerCPUPod as cpuPrice
+			pricePerMemoryPod as memoryPrice
+			cpuCost: cpuCostPod as math(cpuPod * durationInHoursPod * pricePerCPUPod)
+			memoryCost: memoryCostPod as math(memoryPod * durationInHoursPod * pricePerMemoryPod)
+			storageCost: storageCostPod as math(storagePod * durationInHoursPod * 0.00013888888)
+			}
+			name
+			type
+			cpu: sum(val(cpuPod))
+			memory: sum(val(memoryPod))
+			storage: sum(val(storagePod))
+			cpuCost: sum(val(cpuCostPod))
+			memoryCost: sum(val(memoryCostPod))
+			storageCost: sum(val(storageCostPod))
+		}
+	}`
+
+const testQueryForHierarchy = `query {
+		parent(func: has(isNode)) @filter(eq(name, "node-minikube")) {
+			name
+			type
+			children: ~node @filter(has(isPod)) {
+				name
+				type
+			}
+		}
+	}`
+
+const podPriceTestQuery = `query {
+		pod(func: has(isPod)) @filter(eq(name, "pod-purser-dgraph-0")) {
+			cpuPrice
+			memoryPrice
+		}
+	}`
+
+const physicalResourceHierarchyTestQuery = `query {
+			children(func: has(name)) @filter(has(isNode) OR has(isPersistentVolume)) {
+				name
+				type
+			}
+		}`
+
+const logicalResourceHierarchyTestQuery = `query {
+			children(func: has(isNamespace)) {
+				name
+				type
+			}
+		}`
+
+const subcriberRetrievalTestQuery = `query {
+		subscribers(func: has(isSubscriber)) @filter(NOT(has(endTime))) {
+			name
+			Spec {
+				headers
+				url
+			}
+		}
+	}`
+
+const podsWithLabelFilterTestQuery = `query {
+		var(func: has(isLabel)) @filter((eq(key, "k1") AND eq(value, "v1"))) {
+            podUIDs as ~label @filter(has(isPod)) {
+				name
+			}
+		}
+		pods(func: uid(podUIDs)) {
+			uid
+			name
+		}
+	}`
+
+const allLivePodsTestQuery = `query {
+		pods(func: has(isPod)) @filter(NOT has(endTime)) {
+			uid
+			xid
+			name
+		}
+	}`
```

Comment:
Seems ok for the moment but have this cleaned up with a better way to actually call the queries instead of rewriting them again just for the tests.

Reply:
It will help us to know if there is any change in query logic. Tests will fail in such cases letting us know what has changed.